docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -124,51 +132,215 @@ public void ConfigureServices(IServiceCollection services)
+when the type does not match.
+
+In the sample below, only `null` and `string` types need to be handled.
 
 ```csharp
 // In Vector3Type
 public override object ParseValue(object value)
 {
-    return ValueConverter.ConvertTo(value, typeof(Vector3));
+    if (value == null)
+        return null;
+
+    if (value is string vector3InputString)
+    {
+        try
+        {
+            var vector3Parts = vector3InputString.Split(',');
+            var x = float.Parse(vector3Parts[0]);
+            var y = float.Parse(vector3Parts[1]);
+            var z = float.Parse(vector3Parts[2]);
+            return new Vector3(x, y, z);
+        }
+        catch
+        {
+            throw new FormatException($"Failed to parse {nameof(Vector3)} from input '{vector3InputString}'. Input should be a string of three comma-separated floats in X Y Z order, ex. 1.0,2.0,3.0");
+        }
+    }
+
+    return ThrowValueConversionError(value);
 }
 ```
 
-For this call to succeed, a type conversion from `string` to `Vector3` must be registered with the `ValueConverter` class. This can be done anywhere since the API is static. For this example, perform the registration in the schema's constructor.
+### 6. Implement `ScalarGraphType.Serialize` so `Vector3` instances can be sent to the client.
+Keep in mind that a `null` value also required to be handled.
+
+It is recommended that the type of data this method returns match the same type as
+is handled within `ParseValue`. In this case, serialization of `Vector3` is serialized as
+a string, which matches the expected data type within `ParseValue`.
 
 ```csharp
-using GraphQL;
-using GraphQL.Language.AST;
-using GraphQL.Types;
-using StarWars.Types;
-using System;
+// In Vector3Type
+
+public override object Serialize(object value)
+{
+    if (value == null)
+        return null;
+
+    if (value is Vector3 vector3)
+    {
+        return $"{vector3.X},{vector3.Y},{vector3.Z}";
+    }
+
+    return ThrowSerializationError(value);
+}
+```
+
+### 7. Override `ScalarGraphType.ToAST` if necessary.
+
+The infrastructure converts default field values to AST representations during initialization
+in order to verify that the default values are valid within an AST tree. The default implementation
+calls `Serialize` to convert the value to its client-side equivalent, then embeds it into an
+appropriate AST node based on the data type returned from `Serialize`. It is likely you will
+only need to override this method if you are creating a custom scalar that returns enumeration
+values, or if you are returning structured data.
 
-public class ExampleSchema : Schema
+### 8. Code-first: Register the .NET type mapping with the schema (optional).
+
+In order to allow simple field mappings like `Field(x => x.Vector)`, you need to register the
+custom scalar with the schema so that the proper graph type can be selected for the field.
+Call `RegisterTypeMapping` with the CLR data model type and the graph type you want it mapped to.
```

Comment:
```suggestion
Call `RegisterTypeMapping` with the CLR data model type and the graph type you want it map to.
```

?

Reply:
No. "..it mapped to" is correct.  "it to map to" would also be correct, but "it map to" is not.

Reply:
ok